lib/inspec/file_provider.rb

```diff
@@ -0,0 +1,219 @@
+# encoding: utf-8
+require 'rubygems/package'
+require 'zlib'
+require 'zip'
+
+module Inspec
+  class FileProvider
+    def self.for_path(path)
+      if path.is_a?(Hash)
```

Comment:
I'd like to leave this simple constructor for now since it is unlikely that we are going to support more archive types anytime soon.